### PR TITLE
Computer player

### PR DESCRIPTION
**Pull request by minaccchiang**
1. change the method findTile in LetterRack class public 

2. working on makeMove method, but have some question with the last step (Error of Non-static method cannot be referenced from a static context)

### CODE REVIEW

**Comment by petesaiacct** [comment]
I think you need to pull and merge the latest Main head and rebase it on to your version then push again. As there are conflicts. So just cancel this pull request. Pull and merge main over your branch or rebase main into your branch. Then submit a new pull request.

**Comment by minaccchiang** [comment]
Can you leave it like this first, I do not quite understand what to do since I thought I have updated my version to the latest one.

**Review by petesaiacct — APPROVED** [review]
This looks good. For future pushes or a later edit, the documentation will need to be included.

See no issues that will conflict or cause problems.

File: src/main/java/CoreEntities/Player/Player.java [hunk 1]
```diff
@@ -20,7 +20,7 @@ public Player(String name, LetterRack rack, int score) {
         this.rack = rack;
         this.score = score;
     }
-    abstract boolean makeMove(String word);
+    abstract boolean makeMove(String word, int start_x, int end_x, int start_y, int end_y);
```
Comment (petesaiacct) [inline src/main/java/CoreEntities/Player/Player.java:23]: I'm wondering if you should abstract a method with this many parameters. If something has that many specific values. I think implementing an interface in a specific class that meets what the boolean needs to check?

File: src/main/java/CoreEntities/Player/ComputerPlayer.java [hunk 2]
```diff
@@ -1,14 +1,46 @@
...
 
-import core_entities.game_parts.LetterRack;
+import core_entities.game_parts.*;
 
 public class ComputerPlayer extends Player{
     public ComputerPlayer(String name, LetterRack rack, int score) {
         super(name, rack, score);
     }
 
+    /**
```
Comment (petesaiacct) [inline src/main/java/CoreEntities/Player/ComputerPlayer.java:10]: This is useful, and makes sense.

File: src/main/java/CoreEntities/Player/Player.java (hunk 1 above)
Comment (minaccchiang) [reply src/main/java/CoreEntities/Player/Player.java:23]: I do not think so. Since it is computer player, this method will only be called by generateMove

**Review by ethanhonest — COMMENTED** [review]
Line 41 should read be GameState.getBoard().placeTiles(tileList, c1, c2). Once that is changed, I'll approve these changes

**Review by ethanhonest — DISMISSED** [review]
Looks good, made the changes that I requested earlier

**Review by JosephSun2003 — DISMISSED** [review]
Should work and modification to LetterRack doesn't change much.

**Review by mhabibi9 — APPROVED** [review]

**Review by patelhrid — APPROVED** [review]
nice

**Pull request merged**